### PR TITLE
new structure for client config and missing features

### PR DESCRIPTION
**Pull request by pkasprzyk**

### CODE REVIEW

File: neptune/new/internal/backends/hosted_neptune_backend.py [hunk 1]
```diff
@@ -124,9 +125,11 @@ def __init__(
         self.proxies = proxies
         self.missing_features = []
 
-        self._http_client, self._client_config = create_http_client_with_auth(
+        http_client, client_config = create_http_client_with_auth(
```
Comment (Raalsky) [inline neptune/new/internal/backends/hosted_neptune_backend.py:128]: Let's try the syntax with `#type` (it works locally in my PyCharm):
```suggestion+3
        self._http_client, self._client_config = create_http_client_with_auth(
        ...
        )  # type: (RequestsClient, ClientConfig)
```

File: neptune/new/exceptions.py [hunk 2]
```diff
@@ -1004,19 +1004,18 @@ def __init__(self, location: str, namespace: str):
 
 
 class NeptuneFeaturesNotAvailableException(NeptuneException):
```
Comment (Raalsky) [inline neptune/new/exceptions.py:1006]: I think the proper naming is in single form Neptune**Feature**~s~NotAvailableException

File: neptune/new/exceptions.py [hunk 3]
```diff
@@ -1004,19 +1004,18 @@ def __init__(self, location: str, namespace: str):
...
-    def __init__(self, features: List[str]):
+    def __init__(self, missing_feature):
         message = """
 {h1}
 ----NeptuneFeaturesNotAvailableException----------------------------------------------
 {end}
 Following features are not yet supported by the Neptune instance you are using:
```
Comment (Raalsky) [inline neptune/new/exceptions.py:1012]:
```suggestion
Following feature is not yet supported by the Neptune instance you are using:
```

File: neptune/new/internal/backends/hosted_neptune_backend.py (hunk 1 above)
Comment (pkasprzyk) [reply neptune/new/internal/backends/hosted_neptune_backend.py:128]: nice, thanks!

File: neptune/new/exceptions.py [hunk 4]
```diff
@@ -1003,20 +1003,19 @@ def __init__(self, location: str, namespace: str):
...
 {h1}
-----NeptuneFeaturesNotAvailableException----------------------------------------------
+----NeptuneFeatureNotAvailableException----------------------------------------------
 {end}
-Following features are not yet supported by the Neptune instance you are using:
-{features_desc}
+Following feature is not yet supported by the Neptune instance you are using:
+{missing_feature}
 
 An update of the Neptune instance is required in order to use them. Please contact your local Neptune administrator
```
Comment (Raalsky) [inline neptune/new/exceptions.py:1015]: `them` -> `it`

**Review by Raalsky — APPROVED** [review]

**Pull request merged**